app/components/Sidebar/components/DocumentLink.tsx

```diff
@@ -248,12 +250,14 @@ function DocumentLink(
         return;
       }
 
-      if (expanded) {
-        documents.move(item.id, collection.id, node.id, 0);
-        return;
-      }
-
-      documents.move(item.id, collection.id, parentId, index + 1);
+      const parentDocumentId = expanded ? node.id : parentId;
+      const droppedDocumentIndex = expanded ? 0 : index + 1;
+      const document = documents.get(item.id);
+      document?.moveWithUndo(
```

Comment:
I feel like for reordering the toast and undo is probably overkill? Definitely feels more important when re-parenting